lib/components/artist/shows.js

```diff
@@ -7,29 +7,101 @@ const { View, Text, StyleSheet, Image, ListView } = React;
 import Headline from '../text/headline';
 import SerifText from '../text/serif'
 import Show from './show'
+import OpaqueImageView from '../opaque_image_view'
+
+var url = 'https://www.pangeareptile.com/Images/care/baby.jpg'
 
-export default class Shows extends React.Component {
 
-  updateShowData(props) {
-    // this.setState({
-    //   dataSource: this.state.dataSource.cloneWithRows(this.props.show),
-    //   loaded: true,
-    // });
+export default class Shows extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentShowsDataSource: new ListView.DataSource({
+        rowHasChanged:(row1, row2) => row1 !== row2,
+      }),
+      pastShowsDataSource: new ListView.DataSource({
+        rowHasChanged:(row1, row2) => row1 != row2,
+      }),
+      loaded: false,
+    };
   }
 
-  renderShow() {
-    return <ShowView show={this.props.show} />
+  componentDidMount() {
+    this.setState({
```

Comment:
Right, that makes a lot of sense, but what I don’t really understand is why the state needs changing at all? Don’t you know the props during construction time and thus can’t you just setup the appropriate data source there?